src/components/ec-table-filter/ec-table-filter.vue

```diff
@@ -58,12 +63,21 @@ const props = defineProps({
     type: String,
     default: 'Clear filters',
   },
+  isClearButtonHidden: {
+    type: Boolean,
+    default: false,
+  },
 });
 
 const emit = defineEmits(['update:modelValue', 'change']);
 
+const shownFilters = computed(() => (props.filters ? props.filters.filter(filter => !filter.isHidden) : []));
 const hasFilters = computed(() => !!Object.keys(props.modelValue).length);
 
+watch(() => shownFilters.value, () => {
```

Comment:
why we clear filters if change hide or show some?

Reply:
We agreed to clear them when a filter is hidden to avoid having a filter set that cannot be clear by the user. However, now that you mention it maybe it will be a better idea to not clear all of them but clearing only the ones that are going to be hidden...

Reply:
I'd delegate this logic to component user to have flexibility in place where we use filters.

Reply:
mm... tbh I'm not to sure about that, I mean how are we going to clean the filters from other components?

Reply:
We use `ec-smart-table` and pass filters(including `isHidden` flag for each of them). When we decide to hide a filter at the same time we could clear the filter value changing `filterModel`.

Reply:
oh okay! Then I'm going to remove this